src/Framework/IDiagnosticLogger.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Diagnostic information that a build logger cares about.
+    /// </summary>
+    [Flags]
+    public enum DiagnosticInformation
```

Comment:
How comprehensive is this supposed to be?

What about telemetry events? [https://github.com/Microsoft/msbuild/blob/master/src/Framework/TelemetryEventArgs.cs](https://github.com/Microsoft/msbuild/blob/master/src/Framework/TelemetryEventArgs.cs)